### ISSUE TITLE
Custom view onDidChangeTreeData event doesn't refresh label

### ISSUE THREAD

**Issue by EricJizbaMSFT** [microsoft/vscode#78222]
Issue Type: <b>Bug</b>

## Repro steps:
1. Clone ej/treeBug2 branch from this repo (Or see below sample code): https://github.com/EricJizbaMSFT/helloworldext/tree/ej/treeBug2
1. F5 the extension and open the custom view with the beaker icon
1. Right click on '0' node and select "incrementAndRefresh"
![Screen Shot 2019-07-30 at 2 34 31 PM](https://user-images.githubusercontent.com/11282622/62167051-2df11d80-b2d7-11e9-81f1-7ec0c9f2b8b4.png)

### Expected:
The label now shows 1. (This is what happens on VS Code stable v1.36.1)

### Actual:
The label still shows 0. However, if I right click "refreshWholeTree" it will update correctly.

## More details
This bug was originally for the azure functions extension (see https://github.com/microsoft/vscode-azurefunctions/issues/1375), but I tried to create a simple example. It seems like the label is no longer updated when you pass an object to `_onDidChangeTreeData.fire` that has an id.

## Sample code:
```typescript
'use strict';

import * as vscode from 'vscode';

export function activate(context: vscode.ExtensionContext) {
    const tree = new HelloWorldTree();
    context.subscriptions.push(vscode.commands.registerCommand('extension.incrementAndRefresh', async (node: Node) => {
        node.counter += 1;
        tree._onDidChangeTreeData.fire(node);
    }));

    context.subscriptions.push(vscode.commands.registerCommand('extension.refreshWholeTree', async (node: Node) => {
        tree._onDidChangeTreeData.fire();
    }));

    context.subscriptions.push(vscode.window.registerTreeDataProvider('helloworldtree1', tree));
}

export function deactivate() {
}

class HelloWorldTree implements vscode.TreeDataProvider<Node> {
    _onDidChangeTreeData: vscode.EventEmitter<Node> = new vscode.EventEmitter<Node>();
    onDidChangeTreeData: vscode.Event<Node> = this._onDidChangeTreeData.event;

    node1: Node = new Node('1');

    getTreeItem(element: Node): vscode.TreeItem {
        return element;
    }

    async getChildren(element?: Node | undefined): Promise<Node[]> {
        return [this.node1];
    }
}

class Node implements vscode.TreeItem {
    public counter: number = 0;

    constructor(public id: string) { }

    get label(): string {
        return this.counter.toString();
    }
}
```

VS Code version: Code - Insiders 1.37.0-insider (cf03ea3729dc37e439d724267ad29dc5840f903a, 2019-07-30T07:10:53.695Z)
OS version: Darwin x64 18.7.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 2, 2|
|Memory (System)|16.00GB (4.08GB free)|
|Process Argv|.|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (13)</summary>

Extension|Author (truncated)|Version
---|---|---
azurerm-vscode-tools|ms-|0.6.1
vscode-azureappservice|ms-|0.15.0
python|ms-|2019.6.24221
remote-containers|ms-|0.67.0
remote-ssh|ms-|0.44.2
remote-ssh-edit|ms-|0.44.2
remote-ssh-explorer|ms-|0.44.2
azure-account|ms-|0.8.4
csharp|ms-|1.21.0
vscode-typescript-tslint-plugin|ms-|1.2.2
java|red|0.47.0
sort-lines|Tyr|1.8.0
vscode-java-debug|vsc|0.20.0

</details>

**Comment by alexr00** [comment]
Thank you for the easy to use sample! I could repro with the version you're on. This is fixed with 3b6c226be170b60f846d63233da2c6e511f187f0.

**Comment by EricJizbaMSFT** [comment]
Looks good now, thanks!
